src/fiskaltrust.Launcher.Common/Configuration/Converters.cs

```diff
@@ -3,10 +3,28 @@
 
 namespace fiskaltrust.Launcher.Common.Configuration
 {
-    public class SemVersionConverter : JsonConverter<SemanticVersioning.Range>
+    public class SemVersionConverter : JsonConverter<SemanticVersioning.Range?>
     {
-        public override SemanticVersioning.Range Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) => new(reader.GetString());
-
-        public override void Write(Utf8JsonWriter writer, SemanticVersioning.Range semVersionValue, JsonSerializerOptions options) => writer.WriteStringValue(semVersionValue.ToString());
+        public override SemanticVersioning.Range? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var value = reader.GetString();
+            if (string.IsNullOrEmpty(value))
+            {
+                return null;
+            }
+            return new(value);
+        }
+        public override void Write(Utf8JsonWriter writer, SemanticVersioning.Range? semVersionValue, JsonSerializerOptions options)
+        {
+            var value = semVersionValue?.ToString();
+            if (string.IsNullOrEmpty(value))
+            {
+                writer.WriteNullValue();
+            }
+            else
+            {
+                writer.WriteRawValue($"\"{value}\"");
```

Comment:
the `Write` method in the `SemVersionConverter` class, manually formatting the semVersionValue as a JSON string. While the current implementation using `WriteRawValue` should work properly since we manually add quotes around the value, but using the `[WriteStringValue]( https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.writestringvalue?view=net-6.0
)` method could be more error-proof (?). `WriteStringValue` automatically handles proper escaping of special characters

Reply:
```writer.WriteStringValue(value);
```

Reply:
`writer.WriteStringValue(value);`

Reply:
We're actually [using unsafe encoding](https://github.com/fiskaltrust/middleware-launcher/blob/main/src/fiskaltrust.Launcher.Common/Configuration/Configuration.cs#L218) when serializing the config but you are of course correct that using `WriteRawValue` is not a good idea😁

The thing I was trying to fix (the Version `">= 2.0.0"` being serialized as `"\u003E= 2.0.0"`) was actually caused by us not using the unsafe encoding [here](https://github.com/fiskaltrust/middleware-launcher/pull/131/commits/7131a79e4b4aaf87bd08e6a1c00314aec76b7e89#diff-7cf23f76d214e4aa9fa72a6e4d1f51df8cb4a454bf37184e999413293cef0811R141)

Reply:
Thanks for explanation @volllly